src/App.tsx

```diff
@@ -14,11 +14,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+interface State {
+  originGoods: string[],
+  goods: string[],
+  isVisible: boolean,
+  isReverse: boolean,
+  isSortByAlphabet: boolean,
+  isSortByLength: boolean,
+}
+
+export default class App extends React.Component<{}, State> {
+  state:State = {
+    originGoods: [...goodsFromServer],
+    goods: goodsFromServer,
+    isVisible: false,
+    isReverse: false,
+    isSortByAlphabet: false,
+    isSortByLength: false,
+  };
+
+  toggleVisible = () => this.setState(({ isVisible }) => ({ isVisible: !isVisible }));
+
+  toggleReverse = () => this.setState(({ isReverse }) => ({
+    isReverse: !isReverse,
+    goods: goodsFromServer.reverse(),
+  }));
+
+  sortByAlphabet = () => {
+    this.setState(({ isSortByAlphabet }) => ({
+      isSortByAlphabet: !isSortByAlphabet,
+      goods: goodsFromServer.sort((f:string, s:string) => f.localeCompare(s)),
+    }));
+  };
+
+  sortByLength = () => this.setState(({ isSortByLength }) => ({
+    isSortByLength: !isSortByLength,
+    goods: goodsFromServer.sort((a:string, b:string) => a.length - b.length),
```

Comment:
sort without compare function should work the same

```suggestion
    goods: goodsFromServer.sort(),
```